ipn/ipnlocal: TestOnTailnetDefaultAutoUpdate fails on macOS

@awly, #11894 is back. TestOnTailnetDefaultAutoUpdate fails again on macOS.

At 21400756a0a6d3 ...

```
$ go test -v -run=TestOnTailnetDefaultAutoUpdate ./ipn/ipnlocal

=== RUN   TestOnTailnetDefaultAutoUpdate
=== RUN   TestOnTailnetDefaultAutoUpdate/before=,after=true
    logger.go:105:     ... ... [v1] using fake (no-op) tun device
    logger.go:105:     ... ... [v1] using fake (no-op) OS network configurator
    logger.go:105:     ... ... [v1] using fake (no-op) DNS configurator
    logger.go:105:     ... ... dns: using dns.noopManager
    logger.go:105:     ... ... link state: interfaces.State{defaultRoute=en0 ifs={en0:[192.168.1.134/24 2603:8000:8300:6d4::1aa5/64 2603:8000:8300:6d4:10cb:5439:769f:83dc/64 2603:8000:8300:6d4:d58e:9034:2609:7cd8/64 fd00:147d:582:c7bc:1489:1083:afb5:ccec/64 llu6] utun6:[100.77.140.60/32 fd7a:115c:a1e0::4301:8c3b/48 llu6]} v4=true v6=true}
    logger.go:105:     ... ... onPortUpdate(port=51513, network=udp6)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... onPortUpdate(port=60598, network=udp4)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... [v1] magicsock: peermtu: peer MTU status is false
    logger.go:105:     ... ... magicsock: disco key = d:2c70dfc5bebec579
    logger.go:105:     ... ... Creating WireGuard device...
    logger.go:105:     ... ... Bringing WireGuard device up...
    logger.go:105:     ... ... wg: [v2] UDP bind has been updated
    logger.go:105:     ... ... wg: [v2] Interface state was Down, requested Up, now Up
    logger.go:105:     ... ... Bringing router up...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Up: not implemented.
    logger.go:105:     ... ... Clearing router settings...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Set: not implemented.
    logger.go:105:     ... ... Starting network monitor...
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - started
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... Engine created.
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... logpolicy: using system state directory "/Users/bradfitz/.local/share/tailscale"
    logger.go:105:     ... ... [v1] netmap packet filter: (not ready yet)
    logger.go:105:     ... ... ipnext: active extensions: 
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... [v1] warning: fakeRouter.Close: not implemented.
    logger.go:105:     ... ... wg: [v2] Device closing
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - stopped
    logger.go:105:     ... ... wg: [v2] Device closed
=== RUN   TestOnTailnetDefaultAutoUpdate/before=,after=false
    logger.go:105:     ... ... [v1] using fake (no-op) tun device
    logger.go:105:     ... ... [v1] using fake (no-op) OS network configurator
    logger.go:105:     ... ... [v1] using fake (no-op) DNS configurator
    logger.go:105:     ... ... dns: using dns.noopManager
    logger.go:105:     ... ... link state: interfaces.State{defaultRoute=en0 ifs={en0:[192.168.1.134/24 2603:8000:8300:6d4::1aa5/64 2603:8000:8300:6d4:10cb:5439:769f:83dc/64 2603:8000:8300:6d4:d58e:9034:2609:7cd8/64 fd00:147d:582:c7bc:1489:1083:afb5:ccec/64 llu6] utun6:[100.77.140.60/32 fd7a:115c:a1e0::4301:8c3b/48 llu6]} v4=true v6=true}
    logger.go:105:     ... ... onPortUpdate(port=57172, network=udp6)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... onPortUpdate(port=52387, network=udp4)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... [v1] magicsock: peermtu: peer MTU status is false
    logger.go:105:     ... ... magicsock: disco key = d:4f1fb9e9c1b4ef60
    logger.go:105:     ... ... Creating WireGuard device...
    logger.go:105:     ... ... Bringing WireGuard device up...
    logger.go:105:     ... ... wg: [v2] UDP bind has been updated
    logger.go:105:     ... ... wg: [v2] Interface state was Down, requested Up, now Up
    logger.go:105:     ... ... Bringing router up...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Up: not implemented.
    logger.go:105:     ... ... Clearing router settings...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Set: not implemented.
    logger.go:105:     ... ... Starting network monitor...
    logger.go:105:     ... ... Engine created.
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - started
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... logpolicy: using system state directory "/Users/bradfitz/.local/share/tailscale"
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... [v1] netmap packet filter: (not ready yet)
    logger.go:105:     ... ... ipnext: active extensions: 
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    local_test.go:2672: got: "", want "false"
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... [v1] warning: fakeRouter.Close: not implemented.
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... wg: [v2] Device closing
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - stopped
    logger.go:105:     ... ... wg: [v2] Device closed
=== RUN   TestOnTailnetDefaultAutoUpdate/before=unset,after=true
    logger.go:105:     ... ... [v1] using fake (no-op) tun device
    logger.go:105:     ... ... [v1] using fake (no-op) OS network configurator
    logger.go:105:     ... ... [v1] using fake (no-op) DNS configurator
    logger.go:105:     ... ... dns: using dns.noopManager
    logger.go:105:     ... ... link state: interfaces.State{defaultRoute=en0 ifs={en0:[192.168.1.134/24 2603:8000:8300:6d4::1aa5/64 2603:8000:8300:6d4:10cb:5439:769f:83dc/64 2603:8000:8300:6d4:d58e:9034:2609:7cd8/64 fd00:147d:582:c7bc:1489:1083:afb5:ccec/64 llu6] utun6:[100.77.140.60/32 fd7a:115c:a1e0::4301:8c3b/48 llu6]} v4=true v6=true}
    logger.go:105:     ... ... onPortUpdate(port=52153, network=udp6)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... onPortUpdate(port=54741, network=udp4)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... [v1] magicsock: peermtu: peer MTU status is false
    logger.go:105:     ... ... magicsock: disco key = d:35b73ebfab5cf684
    logger.go:105:     ... ... Creating WireGuard device...
    logger.go:105:     ... ... Bringing WireGuard device up...
    logger.go:105:     ... ... wg: [v2] UDP bind has been updated
    logger.go:105:     ... ... wg: [v2] Interface state was Down, requested Up, now Up
    logger.go:105:     ... ... Bringing router up...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Up: not implemented.
    logger.go:105:     ... ... Clearing router settings...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Set: not implemented.
    logger.go:105:     ... ... Starting network monitor...
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... Engine created.
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - started
    logger.go:105:     ... ... logpolicy: using system state directory "/Users/bradfitz/.local/share/tailscale"
    logger.go:105:     ... ... [v1] netmap packet filter: (not ready yet)
    logger.go:105:     ... ... ipnext: active extensions: 
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... [v1] warning: fakeRouter.Close: not implemented.
    logger.go:105:     ... ... wg: [v2] Device closing
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - stopped
    logger.go:105:     ... ... wg: [v2] Device closed
=== RUN   TestOnTailnetDefaultAutoUpdate/before=unset,after=false
    logger.go:105:     ... ... [v1] using fake (no-op) tun device
    logger.go:105:     ... ... [v1] using fake (no-op) OS network configurator
    logger.go:105:     ... ... [v1] using fake (no-op) DNS configurator
    logger.go:105:     ... ... dns: using dns.noopManager
    logger.go:105:     ... ... link state: interfaces.State{defaultRoute=en0 ifs={en0:[192.168.1.134/24 2603:8000:8300:6d4::1aa5/64 2603:8000:8300:6d4:10cb:5439:769f:83dc/64 2603:8000:8300:6d4:d58e:9034:2609:7cd8/64 fd00:147d:582:c7bc:1489:1083:afb5:ccec/64 llu6] utun6:[100.77.140.60/32 fd7a:115c:a1e0::4301:8c3b/48 llu6]} v4=true v6=true}
    logger.go:105:     ... ... onPortUpdate(port=63881, network=udp6)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... onPortUpdate(port=54682, network=udp4)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... [v1] magicsock: peermtu: peer MTU status is false
    logger.go:105:     ... ... magicsock: disco key = d:7489672bedaa62f6
    logger.go:105:     ... ... Creating WireGuard device...
    logger.go:105:     ... ... Bringing WireGuard device up...
    logger.go:105:     ... ... wg: [v2] UDP bind has been updated
    logger.go:105:     ... ... wg: [v2] Interface state was Down, requested Up, now Up
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... Bringing router up...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Up: not implemented.
    logger.go:105:     ... ... Clearing router settings...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Set: not implemented.
    logger.go:105:     ... ... Starting network monitor...
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - started
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... Engine created.
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... logpolicy: using system state directory "/Users/bradfitz/.local/share/tailscale"
    logger.go:105:     ... ... [v1] netmap packet filter: (not ready yet)
    logger.go:105:     ... ... ipnext: active extensions: 
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    local_test.go:2672: got: "unset", want "false"
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... [v1] warning: fakeRouter.Close: not implemented.
    logger.go:105:     ... ... wg: [v2] Device closing
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - stopped
    logger.go:105:     ... ... wg: [v2] Device closed
=== RUN   TestOnTailnetDefaultAutoUpdate/before=false,after=false
    logger.go:105:     ... ... [v1] using fake (no-op) tun device
    logger.go:105:     ... ... [v1] using fake (no-op) OS network configurator
    logger.go:105:     ... ... [v1] using fake (no-op) DNS configurator
    logger.go:105:     ... ... dns: using dns.noopManager
    logger.go:105:     ... ... link state: interfaces.State{defaultRoute=en0 ifs={en0:[192.168.1.134/24 2603:8000:8300:6d4::1aa5/64 2603:8000:8300:6d4:10cb:5439:769f:83dc/64 2603:8000:8300:6d4:d58e:9034:2609:7cd8/64 fd00:147d:582:c7bc:1489:1083:afb5:ccec/64 llu6] utun6:[100.77.140.60/32 fd7a:115c:a1e0::4301:8c3b/48 llu6]} v4=true v6=true}
    logger.go:105:     ... ... onPortUpdate(port=58026, network=udp6)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... onPortUpdate(port=64069, network=udp4)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... [v1] magicsock: peermtu: peer MTU status is false
    logger.go:105:     ... ... magicsock: disco key = d:495c038231f26563
    logger.go:105:     ... ... Creating WireGuard device...
    logger.go:105:     ... ... Bringing WireGuard device up...
    logger.go:105:     ... ... wg: [v2] UDP bind has been updated
    logger.go:105:     ... ... wg: [v2] Interface state was Down, requested Up, now Up
    logger.go:105:     ... ... Bringing router up...
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... [v1] warning: fakeRouter.Up: not implemented.
    logger.go:105:     ... ... Clearing router settings...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Set: not implemented.
    logger.go:105:     ... ... Starting network monitor...
    logger.go:105:     ... ... Engine created.
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - started
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... logpolicy: using system state directory "/Users/bradfitz/.local/share/tailscale"
    logger.go:105:     ... ... [v1] netmap packet filter: (not ready yet)
    logger.go:105:     ... ... ipnext: active extensions: 
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... [v1] warning: fakeRouter.Close: not implemented.
    logger.go:105:     ... ... wg: [v2] Device closing
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - stopped
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... wg: [v2] Device closed
=== RUN   TestOnTailnetDefaultAutoUpdate/before=true,after=true
    logger.go:105:     ... ... [v1] using fake (no-op) tun device
    logger.go:105:     ... ... [v1] using fake (no-op) OS network configurator
    logger.go:105:     ... ... [v1] using fake (no-op) DNS configurator
    logger.go:105:     ... ... dns: using dns.noopManager
    logger.go:105:     ... ... link state: interfaces.State{defaultRoute=en0 ifs={en0:[192.168.1.134/24 2603:8000:8300:6d4::1aa5/64 2603:8000:8300:6d4:10cb:5439:769f:83dc/64 2603:8000:8300:6d4:d58e:9034:2609:7cd8/64 fd00:147d:582:c7bc:1489:1083:afb5:ccec/64 llu6] utun6:[100.77.140.60/32 fd7a:115c:a1e0::4301:8c3b/48 llu6]} v4=true v6=true}
    logger.go:105:     ... ... onPortUpdate(port=60408, network=udp6)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... onPortUpdate(port=65119, network=udp4)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... [v1] magicsock: peermtu: peer MTU status is false
    logger.go:105:     ... ... magicsock: disco key = d:470e8b277cade7a2
    logger.go:105:     ... ... Creating WireGuard device...
    logger.go:105:     ... ... Bringing WireGuard device up...
    logger.go:105:     ... ... wg: [v2] UDP bind has been updated
    logger.go:105:     ... ... wg: [v2] Interface state was Down, requested Up, now Up
    logger.go:105:     ... ... Bringing router up...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Up: not implemented.
    logger.go:105:     ... ... Clearing router settings...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Set: not implemented.
    logger.go:105:     ... ... Starting network monitor...
    logger.go:105:     ... ... Engine created.
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - started
    logger.go:105:     ... ... logpolicy: using system state directory "/Users/bradfitz/.local/share/tailscale"
    logger.go:105:     ... ... [v1] netmap packet filter: (not ready yet)
    logger.go:105:     ... ... ipnext: active extensions: 
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... [v1] warning: fakeRouter.Close: not implemented.
    logger.go:105:     ... ... wg: [v2] Device closing
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - stopped
    logger.go:105:     ... ... wg: [v2] Device closed
=== RUN   TestOnTailnetDefaultAutoUpdate/before=,after=
    logger.go:105:     ... ... [v1] using fake (no-op) tun device
    logger.go:105:     ... ... [v1] using fake (no-op) OS network configurator
    logger.go:105:     ... ... [v1] using fake (no-op) DNS configurator
    logger.go:105:     ... ... dns: using dns.noopManager
    logger.go:105:     ... ... link state: interfaces.State{defaultRoute=en0 ifs={en0:[192.168.1.134/24 2603:8000:8300:6d4::1aa5/64 2603:8000:8300:6d4:10cb:5439:769f:83dc/64 2603:8000:8300:6d4:d58e:9034:2609:7cd8/64 fd00:147d:582:c7bc:1489:1083:afb5:ccec/64 llu6] utun6:[100.77.140.60/32 fd7a:115c:a1e0::4301:8c3b/48 llu6]} v4=true v6=true}
    logger.go:105:     ... ... onPortUpdate(port=65376, network=udp6)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... onPortUpdate(port=65352, network=udp4)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... [v1] magicsock: peermtu: peer MTU status is false
    logger.go:105:     ... ... magicsock: disco key = d:cfdf4e7425f3f207
    logger.go:105:     ... ... Creating WireGuard device...
    logger.go:105:     ... ... Bringing WireGuard device up...
    logger.go:105:     ... ... wg: [v2] UDP bind has been updated
    logger.go:105:     ... ... wg: [v2] Interface state was Down, requested Up, now Up
    logger.go:105:     ... ... Bringing router up...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Up: not implemented.
    logger.go:105:     ... ... Clearing router settings...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Set: not implemented.
    logger.go:105:     ... ... Starting network monitor...
    logger.go:105:     ... ... Engine created.
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - started
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... logpolicy: using system state directory "/Users/bradfitz/.local/share/tailscale"
    logger.go:105:     ... ... [v1] netmap packet filter: (not ready yet)
    logger.go:105:     ... ... ipnext: active extensions: 
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... [v1] warning: fakeRouter.Close: not implemented.
    logger.go:105:     ... ... wg: [v2] Device closing
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - stopped
    logger.go:105:     ... ... wg: [v2] Device closed
=== RUN   TestOnTailnetDefaultAutoUpdate/before=false,after=false#01
    logger.go:105:     ... ... [v1] using fake (no-op) tun device
    logger.go:105:     ... ... [v1] using fake (no-op) OS network configurator
    logger.go:105:     ... ... [v1] using fake (no-op) DNS configurator
    logger.go:105:     ... ... dns: using dns.noopManager
    logger.go:105:     ... ... link state: interfaces.State{defaultRoute=en0 ifs={en0:[192.168.1.134/24 2603:8000:8300:6d4::1aa5/64 2603:8000:8300:6d4:10cb:5439:769f:83dc/64 2603:8000:8300:6d4:d58e:9034:2609:7cd8/64 fd00:147d:582:c7bc:1489:1083:afb5:ccec/64 llu6] utun6:[100.77.140.60/32 fd7a:115c:a1e0::4301:8c3b/48 llu6]} v4=true v6=true}
    logger.go:105:     ... ... onPortUpdate(port=63513, network=udp6)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... onPortUpdate(port=51298, network=udp4)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... [v1] magicsock: peermtu: peer MTU status is false
    logger.go:105:     ... ... magicsock: disco key = d:b682579b62c5ea22
    logger.go:105:     ... ... Creating WireGuard device...
    logger.go:105:     ... ... Bringing WireGuard device up...
    logger.go:105:     ... ... wg: [v2] UDP bind has been updated
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - started
    logger.go:105:     ... ... wg: [v2] Interface state was Down, requested Up, now Up
    logger.go:105:     ... ... Bringing router up...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Up: not implemented.
    logger.go:105:     ... ... Clearing router settings...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Set: not implemented.
    logger.go:105:     ... ... Starting network monitor...
    logger.go:105:     ... ... Engine created.
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... logpolicy: using system state directory "/Users/bradfitz/.local/share/tailscale"
    logger.go:105:     ... ... [v1] netmap packet filter: (not ready yet)
    logger.go:105:     ... ... ipnext: active extensions: 
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... [v1] warning: fakeRouter.Close: not implemented.
    logger.go:105:     ... ... wg: [v2] Device closing
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - stopped
    logger.go:105:     ... ... wg: [v2] Device closed
=== RUN   TestOnTailnetDefaultAutoUpdate/before=true,after=true#01
    logger.go:105:     ... ... [v1] using fake (no-op) tun device
    logger.go:105:     ... ... [v1] using fake (no-op) OS network configurator
    logger.go:105:     ... ... [v1] using fake (no-op) DNS configurator
    logger.go:105:     ... ... dns: using dns.noopManager
    logger.go:105:     ... ... link state: interfaces.State{defaultRoute=en0 ifs={en0:[192.168.1.134/24 2603:8000:8300:6d4::1aa5/64 2603:8000:8300:6d4:10cb:5439:769f:83dc/64 2603:8000:8300:6d4:d58e:9034:2609:7cd8/64 fd00:147d:582:c7bc:1489:1083:afb5:ccec/64 llu6] utun6:[100.77.140.60/32 fd7a:115c:a1e0::4301:8c3b/48 llu6]} v4=true v6=true}
    logger.go:105:     ... ... onPortUpdate(port=59084, network=udp6)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... onPortUpdate(port=52327, network=udp4)
    logger.go:105:     ... ... [v1] warning: fakeRouter.UpdateMagicsockPort: not implemented.
    logger.go:105:     ... ... [v1] magicsock: peermtu: peer MTU status is false
    logger.go:105:     ... ... magicsock: disco key = d:5b68f4987906e396
    logger.go:105:     ... ... Creating WireGuard device...
    logger.go:105:     ... ... Bringing WireGuard device up...
    logger.go:105:     ... ... wg: [v2] UDP bind has been updated
    logger.go:105:     ... ... wg: [v2] Interface state was Down, requested Up, now Up
    logger.go:105:     ... ... Bringing router up...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Up: not implemented.
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... Clearing router settings...
    logger.go:105:     ... ... [v1] warning: fakeRouter.Set: not implemented.
    logger.go:105:     ... ... Starting network monitor...
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - started
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - started
    logger.go:105:     ... ... Engine created.
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... logpolicy: using system state directory "/Users/bradfitz/.local/share/tailscale"
    logger.go:105:     ... ... [v1] netmap packet filter: (not ready yet)
    logger.go:105:     ... ... ipnext: active extensions: 
    logger.go:105:     ... ... pm: migrating "_daemon" profile to new format
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming mkReceiveFunc - stopped
    logger.go:105:     ... ... [v1] warning: fakeRouter.Close: not implemented.
    logger.go:105:     ... ... wg: [v2] Device closing
    logger.go:105:     ... ... wg: [v2] Routine: receive incoming receiveDERP - stopped
    logger.go:105:     ... ... wg: [v2] Device closed
--- FAIL: TestOnTailnetDefaultAutoUpdate (0.02s)
    --- PASS: TestOnTailnetDefaultAutoUpdate/before=,after=true (0.01s)
    --- FAIL: TestOnTailnetDefaultAutoUpdate/before=,after=false (0.00s)
    --- PASS: TestOnTailnetDefaultAutoUpdate/before=unset,after=true (0.00s)
    --- FAIL: TestOnTailnetDefaultAutoUpdate/before=unset,after=false (0.00s)
    --- PASS: TestOnTailnetDefaultAutoUpdate/before=false,after=false (0.00s)
    --- PASS: TestOnTailnetDefaultAutoUpdate/before=true,after=true (0.00s)
    --- PASS: TestOnTailnetDefaultAutoUpdate/before=,after= (0.00s)
    --- PASS: TestOnTailnetDefaultAutoUpdate/before=false,after=false#01 (0.00s)
    --- PASS: TestOnTailnetDefaultAutoUpdate/before=true,after=true#01 (0.00s)
FAIL
FAIL	tailscale.com/ipn/ipnlocal	0.304s
FAIL
```

--- a/ipn/ipnlocal/local_test.go
+++ b/ipn/ipnlocal/local_test.go
@@ -15,7 +15,6 @@ import (
 	"os"
 	"path/filepath"
 	"reflect"
-	"runtime"
 	"slices"
 	"strings"
 	"sync"
@@ -2597,9 +2596,6 @@ func TestPreferencePolicyInfo(t *testing.T) {
 }
 
 func TestOnTailnetDefaultAutoUpdate(t *testing.T) {
-	if runtime.GOOS == "darwin" {
-		t.Skip("test known broken on macOS; see https://github.com/tailscale/tailscale/issues/15691")
-	}
 	tests := []struct {
 		before, after  opt.Bool
 		container      opt.Bool
@@ -2669,7 +2665,7 @@ func TestOnTailnetDefaultAutoUpdate(t *testing.T) {
 			// On platforms that don't support auto-update we can never
 			// transition to auto-updates being enabled. The value should
 			// remain unchanged after onTailnetDefaultAutoUpdate.
-			if !clientupdate.CanAutoUpdate() && want.EqualBool(true) {
+			if !clientupdate.CanAutoUpdate() {
 				want = tt.before
 			}
 			if got := b.pm.CurrentPrefs().AutoUpdate().Apply; got != want {

